nixos/modules/programs/foot/default.nix

```
@@ -0,0 +1,92 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.foot;
+
+  settingsFormat = pkgs.formats.ini {
+    listsAsDuplicateKeys = true;
+    mkKeyValue =
+      with lib.generators;
+      mkKeyValueDefault {
+        mkValueString =
+          v:
+          mkValueStringDefault { } (
+            if v == true then
+              "yes"
+            else if v == false then
+              "no"
+            else if v == null then
+              "none"
+            else
+              v
+          );
+      } "=";
+  };
+in
+{
+  options.programs.foot = {
+    enable = lib.mkEnableOption "foot terminal emulator";
+
+    package = lib.mkPackageOption pkgs "foot" { };
+
+    settings = lib.mkOption {
+      inherit (settingsFormat) type;
+      default = { };
+      description = ''
+        Configuration for foot terminal emulator. Further information can be found in {command}`man 5 foot.ini`.
+      '';
+      example = {
+        scrollback.lines = 100000;
+      };
+    };
+
+    theme = lib.mkOption {
+      type = with lib.types; nullOr str;
+      default = null;
+      description = ''
+        Theme name. Check <https://codeberg.org/dnkl/foot/src/branch/master/themes> for available themes.
+      '';
+      example = "aeroroot";
+    };
+
+    enableBashIntegration = lib.mkEnableOption "foot bash integration" // {
+      enable = true;
```

same for fish and zsh
```suggestion
      default = true;
```